### PR TITLE
fix: Rename MYPY to TYPE_CHECKING

### PR DESCRIPTION
**Pull request by untitaker**
we have a lot of conditionals in our codebase that are supposed to
separate the code that mypy is supposed to see from the code that we
actually want to execute.

In the specific case of sentry_sdk.configure_scope, this means that
pyright does not handle with the overloads correctly because it only
recognizes TYPE_CHECKING as a special variable name, not MYPY.

Rename MYPY to TYPE_CHECKING so pyright typechecks configure_scope
correctly.

### CODE REVIEW

File: scripts/init_serverless_sdk.py [hunk 1]
```diff
@@ -10,11 +10,11 @@
 import re
 
 import sentry_sdk
-from sentry_sdk._types import MYPY
+from sentry_sdk._types import TYPE_CHECKING
 from sentry_sdk.utils import Dsn
 from sentry_sdk.integrations.aws_lambda import AwsLambdaIntegration
 
-if MYPY:
+if TYPE_CHECKING:
```
Comment (sl0thentr0py) [inline scripts/init_serverless_sdk.py:17]: should we still expose `MYPY` as an alias just in case someone's importing it currently?

Comment (untitaker) [reply scripts/init_serverless_sdk.py:17]: oh.... yeah. probably. they shouldn't do that, right?

Comment (sl0thentr0py) [reply scripts/init_serverless_sdk.py:17]: shouldn't.. but I would remove it in a major just to be safe

**Review by sl0thentr0py — APPROVED** [review]

File: sentry_sdk/_types.py [hunk 2]
```diff
@@ -1,10 +1,14 @@
 try:
-    from typing import TYPE_CHECKING as MYPY
+    from typing import TYPE_CHECKING as TYPE_CHECKING
 except ImportError:
-    MYPY = False
+    TYPE_CHECKING = False
```
Comment (aqeelat) [inline sentry_sdk/_types.py:4]: two questions: 
1. why the try-catch?
2. why `from x import y as y`?

**Pull request merged**